bytestring_splitter/__init__.py

```diff
@@ -219,3 +285,35 @@ def __eq__(self, other):
     @classmethod
     def expected_bytes_length(cls):
         return cls
+
+    @classmethod
+    def bundle(cls, collection):
+        """
+        Casts each item in collection to bytes, makes a VariableLengthBytestring from each.
+        Then, casts the result to bytes and makes a VariableLengthBytestring from it.
+
+        Useful for semantically packing collections of individually useful objects without needing boilerplate all over the place.
+        """
+        vbytes = (VariableLengthBytestring(i) for i in collection)
+        concatenated_bytes = bytes().join(bytes(d) for d in vbytes)
+        vbytes_joined = VariableLengthBytestring(concatenated_bytes)
+        return vbytes_joined
+
+    @staticmethod
+    def discharge(bytestring):
+        """
+        Takes a bytestring representation of a VariableLengthBytestring, confirms that it is the correct length,
+        and returns the original bytes.
+        """
+        message_length_as_bytes = bytestring[0:0 + VARIABLE_HEADER_LENGTH]
```

Comment:
just curious: why `[0:0 + VARIABLE_HEADER_LENGTH]`?

Reply:
The first `VARIABLE_HEADER_LENGTH` bytes are the length.

Reply:
Help me make this clearer.

Reply:
why don't just `[:VARIABLE_HEADER_LENGTH]`?

Reply:
Riiiight.